quickstart/aspnet/ContosoTeamStats/RedisConnection.cs

```diff
@@ -142,21 +142,26 @@ private async Task ForceReconnectAsync(bool initializing = false)
                 _firstErrorTime = DateTimeOffset.MinValue;
                 _previousErrorTime = DateTimeOffset.MinValue;
 
-                if (_connection != null)
+                // Create a new connection
+                ConnectionMultiplexer _newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
+
+                // Swap current connection with the new connection
+                ConnectionMultiplexer _oldConnection = Interlocked.Exchange(ref _connection, _newConnection);
+
+                if (_oldConnection != null)
```

Comment:
This `if` block that calls `CloseAsync()` is potentially slow and we don't care when it happens so it should be moved to the end of the method, after updating `_database`

Reply:
moved